pkg/resource/statefulset.go

```diff
@@ -276,7 +276,7 @@ func (b StatefulSetBuilder) dbArgs() []string {
 		"-ecx",
 		">- exec /cockroach/cockroach start" +
 			" --join=" + b.joinStr() +
-			" --advertise-host=$(hostname -f)" +
+			" --advertise-host=$(echo $HOSTNAME)" +
```

Comment:
This may not currently be a concern for the operator, so feel free to push back (but maybe add a note somewhere?)

`hostname -f` is the FQDN while `$HOSTNAME` is just the pod name. This will present issues when looking at multi-region clusters or networking together an operator cluster and a non-operator cluster.

```bash
root@cockroachdb-0:/cockroach# echo $HOSTNAME
cockroachdb-0
root@cockroachdb-0:/cockroach# hostname -f
cockroachdb-0.cockroachdb.us-west-2.svc.cluster.local
```

Reply:
hostname -f does not work on openshift, and `$HOSTNAME` may not work on kind.  E2E with CI failed.  I am currently debugging.